### PR TITLE
New investor UI/UX

### PR DESCRIPTION
**Pull request by AdrielGS**
Changed UI/UX based on: [https://app.atomic.io/d/XcsjpSXbg93d](url)

### CODE REVIEW

File: src/app/common/footer/footer.component.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-footer',
+  templateUrl: './footer.component.html',
+  styleUrls: ['./footer.component.css']
+})
+export class FooterComponent implements OnInit {
+
+  constructor() { }
```
Comment (icaroharry) [inline src/app/common/footer/footer.component.ts]: should this class be empty?

File: src/app/common/services/swapy-protocol.service.ts [hunk 2]
```diff
@@ -48,6 +60,11 @@ export class SwapyProtocolService {
       return origCall.apply(this, arguments);
     };
   }
+  private storeTransactionHash(addresses: string[], hash: string) {
+    addresses.forEach(address => {
+      this.storageService.setItem(address, hash);
```
Comment (icaroharry) [inline src/app/common/services/swapy-protocol.service.ts]: why all the addresses point to the same hash?

File: src/app/common/services/swapy-protocol.service.ts [hunk 3]
```diff
@@ -68,12 +85,12 @@ export class SwapyProtocolService {
...
   }
 
-  private async getEthPrice() {
+  public async getEthPrice() {
     return new Promise((resolve, reject) => {
       this.http.get(this.ethPriceProvider).subscribe(data => {
         resolve((data as any).bid);
       }, error => {
-        resolve(440.0);
+        reject(840);
```
Comment (icaroharry) [inline src/app/common/services/swapy-protocol.service.ts]: I miss this price :laughing: but I think we should throw an error instead

File: src/app/common/services/swapy-protocol.service.ts [hunk 4]
```diff
@@ -212,31 +238,38 @@ export class SwapyProtocolService {
...
     return this.SwapyExchangeContract.methods
-      .sellAsset(contractAddress, value)
+      .sellAssets(contractAddresses, values)
       .send({
         from: this.walletService.getWallet().address,
-        gas: 150000,
+        gas: 150000 * contractAddresses.length,
         gasPrice: this.web3.utils.toWei(this.gasPrice, 'gwei')
       }).on('transactionHash', (hash) => {
         this.handleOnTransactionHash(hash);
```
Comment (icaroharry) [inline src/app/common/services/swapy-protocol.service.ts]: suggestion: create `handleError` to reuse this code

File: src/app/investor/accept-sale/accept-sale.component.css [hunk 5]
```diff
@@ -0,0 +1,163 @@
+.sw-title-outside-box {
```
Comment (icaroharry) [inline src/app/investor/accept-sale/accept-sale.component.css]: css king :crown:

File: src/app/investor/add-funds/add-funds-child/add-funds-child.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (icaroharry) [inline src/app/investor/add-funds/add-funds-child/add-funds-child.component.spec.ts]: :clap: :clap:

tests <3

File: src/app/investor/confirm-purchase/confirm-purchase.component.ts [hunk 7]
```diff
@@ -26,16 +26,36 @@ export class ConfirmPurchaseComponent implements OnInit {
...
+    return monthsDiff;
+  }
+
+  public calculatePaybackDate() {
+    const paybackDate = new Date(this.asset.investedAt);
+    paybackDate.setMonth(paybackDate.getMonth() + this.asset.paybackMonths);
+    return paybackDate;
+  }
+
+  public porcentageProgression() {
```
Comment (icaroharry) [inline src/app/investor/confirm-purchase/confirm-purchase.component.ts]: percentage*

File: src/app/investor/invest/invest.component.html [hunk 8]
```diff
@@ -3,83 +3,117 @@
...
 
     </div>
+      <div class="col-sm-3 col-md-3 sw-company-box">
+        <div class="row">
+          <div class="col-sm-12 col-md-12">
+            <span>Company's wallet address: <p class="sw-blue1">{{investment.walletAddress}}</p></span>
+          </div>
+        </div>
+      <!-- 
+        <div class="row">
```
Comment (icaroharry) [inline src/app/investor/invest/invest.component.html]: remove coments

File: src/app/investor/investor.service.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+import { InvestorComponent } from './investor.component';
+
+@Injectable()
+export class InvestorService {
```
Comment (icaroharry) [inline src/app/investor/investor.service.ts]: maybe delete this service?

File: src/app/investor/message/message.service.ts [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+  public cleanMessages() {
+    this.error = undefined;
+    this.message = undefined;
+    this.headerMessage = undefined;
+    this.loadingLogo = false;
+  }
+
+  public setErrorMessage(message) {
+    if (message !== null) {
```
Comment (icaroharry) [inline src/app/investor/message/message.service.ts]: suggestion: create errorService that handles this messages

File: src/app/common/footer/footer.component.ts (hunk 1 above)
Comment (AdrielGS) [reply src/app/common/footer/footer.component.ts]: Yes, just the HTML and CSS  is required.

File: src/app/common/services/swapy-protocol.service.ts (hunk 2 above)
Comment (AdrielGS) [reply src/app/common/services/swapy-protocol.service.ts]: Let's suppose a user puts 3 assets on sale at the same time. With the new UI (and protocol) the user just needs to send one transaction via MetaMask. The contract internally will do the call for the 3 assets internally, but I just need the hash of the main transaction to see its status later.

File: src/app/common/services/swapy-protocol.service.ts (hunk 3 above)
Comment (AdrielGS) [reply src/app/common/services/swapy-protocol.service.ts]: You are right, I forgot to change this one.

File: src/app/common/services/swapy-protocol.service.ts (hunk 4 above)
Comment (AdrielGS) [reply src/app/common/services/swapy-protocol.service.ts]: Which code specifically?

File: src/app/investor/invest/invest.component.html (hunk 8 above)
Comment (AdrielGS) [reply src/app/investor/invest/invest.component.html]: I'm aware of this. There are loads of comments that should be removed.

File: src/app/investor/investor.service.ts (hunk 9 above)
Comment (AdrielGS) [reply src/app/investor/investor.service.ts]: Deleted :)

**Pull request merged**